WordPress/gutenberg
# Navigation Block : try not using css variable for block gap support

## Description
Trying to refactor navigation block so gap is still supported when [removal of CSS gap variable](https://github.com/WordPress/gutenberg/pull/37360) is merged

Because the elements that need `gap` applied are not the direct children of the block wrapper to which gap is applied after [this refactor](https://github.com/WordPress/gutenberg/pull/37360), this PR instead manually applies the value from the `blockGap` attribute to the relevant navigation item wrappers.

## How has this been tested?

- Add a navigation menu with several items, including some custom links
- Check that it displays as expected in editor and front end
- Apply some block spacing in the right block settings panel and make sure the adding spacing displays as expected in editor  and front end
- Check that disabling `blockGap` in `theme.json` behaves as expected and also setting a default `blockGap` value in `theme.json` 

## Screenshots 
![blockgap](https://user-images.githubusercontent.com/3629020/147032040-2bc94632-2b0d-4b15-9788-45b11d52c17e.gif)
